iceoryx_hoofs/cmake/IceoryxPackageHelper.cmake

```diff
@@ -111,3 +110,281 @@ Macro(install_package_files_and_export)
     DESTINATION ${DESTINATION_CONFIGDIR}
     )
 endMacro()
+
+Macro(iox_make_unique_includedir)
+    option(MAKE_UNIQUE_INCLUDEDIR
+      "When ON headers are installed to a path ending with folders called \
+      iceoryx/vX.Y.Z/ . This avoids include directory search order issues when \
+      overriding this package from a merged catkin, ament, or colcon workspace."
+      ON)
+
+    if(MAKE_UNIQUE_INCLUDEDIR)
+      set(_unique_dir "iceoryx/v${IOX_VERSION_STRING}")
+      if(PREFIX STREQUAL "")
+        set(PREFIX "${_unique_dir}")
+      else()
+        set(PREFIX "${PREFIX}/${_unique_dir}")
+      endif()
+    endif()
+endMacro()
+
+Macro(iox_set_rpath)
+    set(switches IS_EXECUTABLE)
+    set(arguments TARGET)
+    cmake_parse_arguments(IOX "${switches}" "${arguments}" "" ${ARGN} )
+
+    if( APPLE )
+        if ( IOX_IS_EXECUTABLE )
+            set(IOX_RPATH_PREFIX "@executable_path")
+        else()
+            set(IOX_RPATH_PREFIX "@loader_path")
+        endif()
+    elseif ( LINUX OR UNIX )
+        set(IOX_RPATH_PREFIX "\$ORIGIN")
+    endif()
+
+    set_property(
+        TARGET ${IOX_TARGET}
+        PROPERTY INSTALL_RPATH
+        "${IOX_RPATH_PREFIX}/../${CMAKE_INSTALL_LIBDIR}"
+    )
+
+    set_property(
+        TARGET ${IOX_TARGET}
+        PROPERTY BUILD_RPATH
+            "${IOX_RPATH_PREFIX}/../iceoryx_hoofs"
+            "${IOX_RPATH_PREFIX}/../iceoryx_posh"
+            "${IOX_RPATH_PREFIX}/../iceoryx_hoofs/platform"
+            "${IOX_RPATH_PREFIX}/../iceoryx_binding_c"
```

Comment:
Sadly no since colcon uses hoofs and posh directly. In the end we have two options.
1. Quick and simple we keep it as it is for now.
2. Every target has to provide the RPATHS manually, extremely error prone (this was my first approach and I failed).
3. We write an additional function which looks up the included libraries and when they start with `iceoryx_` then an rpath for them will be added. Works with colcon but not in our setup since we build `iceoryx_posh` in `build/posh` instead of `build/iceoryx_posh`.
Of course we could extend this function and remove the `iceoryx_` and add additional RPATHS but this looks even dirtier.

I would propose therefore that we keep this for now like it is. Then we move `build/{posh,dds,binding_c,hoofs}` to `/build/iceoryx_{posh,dds,binding_c,hoofs}` and implement option 3. Then we are compliant again with the good building standards.

But the move will be a big separate PR since I have to adjust then every example. And `iox-roudi` must then also be stored under `./build/iceoryx_posh/iox-roudi`.

Reply:
`iceoryx_meta` is out of the question since colcon builds `iceoryx_posh` and `iceoryx_hoofs` directly and ignores meta.

Another problem are implicit library dependencies. Let's say you build `iox-roudi`, this has a dependency to `iceoryx_posh` which has a dependency to `iceoryx_hoofs` which has a dependency to `iceoryx_platform`.
When `iox-roudi` it wants to load those libraries, but in cmake I do not have direct access to those implicit dependencies easily, see https://stackoverflow.com/questions/32756195/recursive-list-of-link-libraries-in-cmake
But `iox-roudi` requires the correct RPATH for all of them. 
So I could look for all libraries starting with `iceoryx_` and add them to the RPATH but in meta we violate cmake best practices, instead of building `iceoryx_posh` in `build/iceoryx_posh` we build it in `build/posh`.
So we have to handle this exception.

I propose to leave it as it is and solve this issue in a follow up PR.